### PR TITLE
Modification marc21metadata

### PR DESCRIPTION
**Pull request by philippgualdi**
Create Marc21 records allowing other modules use emplace_* methods or load a existing XMLTree object

### CODE REVIEW

File: invenio_records_marc21/services/record/metadata.py [hunk 1]
```diff
@@ -132,13 +142,24 @@ def emplace_unique_field(
...
+            )  # DataField(tag, ind1, ind2)
+        else:
+            datafield = datafield[0]
+        subfield = self._etree.xpath(f".//subfield[@code='{code}']")
+        if not subfield:
+            subfield = etree.Element("subfield", code=code)
+            subfield.text = value
+            datafield.append(subfield)
+            self._etree.append(datafield)
+        pass
```
Comment (utnapischtim) [inline invenio_records_marc21/services/record/metadata.py:162]: minor: remove the pass

File: invenio_records_marc21/services/record/metadata.py [hunk 2]
```diff
@@ -72,45 +77,47 @@ def load(self, xml: etree):
...
-        for df in self.datafields:
-            if (
-                df.tag == ref_df.tag and df.ind1 == ref_df.ind1
-            ) and df.ind2 == ref_df.ind2:
-                for sf in df.subfields:
-                    if sf.code == ref_sf.code and sf.value == ref_sf.value:
-                        return True
+        self._xml = etree.tostring(self._etree, pretty_print=True).decode("UTF-8")
+
+    def contains(self, ref_df: dict, ref_sf: dict):
```
Comment (utnapischtim) [inline invenio_records_marc21/services/record/metadata.py:86]: what means ref_df and ref_sf?

File: invenio_records_marc21/services/services.py [hunk 3]
```diff
@@ -44,8 +44,7 @@ def _create_data(self, identity, data, metadata, files=False, access=None):
         :rtype: dict
         """
         if data is None:
-            record = create_record(metadata.etree)
```
Comment (utnapischtim) [inline invenio_records_marc21/services/services.py:47]: remove create_record from the import list

File: invenio_records_marc21/services/record/metadata.py [hunk 4]
```diff
@@ -14,26 +14,31 @@
 from os import linesep
 from os.path import dirname, join
 
+from dojson.contrib.marc21 import marc21
+from dojson.contrib.marc21.utils import create_record
 from lxml import etree
-
-from .fields import ControlField, DataField, LeaderField, SubField
```
Comment (utnapischtim) [inline invenio_records_marc21/services/record/metadata.py:19]: Are they used somewhere else? maybe we should remove them then completely?

File: invenio_records_marc21/services/record/metadata.py [hunk 5]
```diff
@@ -120,10 +127,13 @@ def emplace_field(
         value: str = "",
     ) -> None:
         """Add value to record for given datafield and subfield."""
-        datafield = DataField(tag, ind1, ind2)
-        subfield = SubField(code, value)
-        datafield.subfields.append(subfield)
-        self.datafields.append(datafield)
+        datafield = etree.Element(
+            "datafield", tag=tag, ind1=ind1, ind2=ind2
+        )  # DataField(tag, ind1, ind2)
```
Comment (utnapischtim) [inline invenio_records_marc21/services/record/metadata.py:132]: Remove the comment. It references an unused and possibly removed class

File: invenio_records_marc21/services/record/metadata.py [hunk 6]
```diff
@@ -132,13 +142,24 @@ def emplace_unique_field(
...
-        if not self.contains(datafield, subfield):
-            datafield.subfields.append(subfield)
-            self.datafields.append(datafield)
+        datafield = self._etree.xpath(
+            f".//datafield[@ind1='{ind1}' and @ind2='{ind2}' and @tag='{tag}']"
+        )
+        if not datafield:
+            datafield = etree.Element(
+                "datafield", tag=tag, ind1=ind1, ind2=ind2
+            )  # DataField(tag, ind1, ind2)
```
Comment (utnapischtim) [inline invenio_records_marc21/services/record/metadata.py:153]: remove the comment

File: invenio_records_marc21/services/record/metadata.py (hunk 2 above)
Comment (philippgualdi) [reply invenio_records_marc21/services/record/metadata.py:86]: @param ref_df dict: datafield element specific information, containing keys [tag,ind1,ind2]
 @param ref_sf dict: subfield element specific information, containing keys [code,value]

**Review by utnapischtim — APPROVED** [review]

**Pull request merged**